### ISSUE TITLE
Data split issue

### ISSUE THREAD

**Issue by zhixinma** [ArrowLuo/CLIP4Clip#81]
Hi @ArrowLuo, I have a question about the data split during the fine-tuning stage. The paper claims that you refer to the data split of the ECCV'20 paper "Multi-modal transformer for video retrieval". In the ECCV'20 paper's [code](https://github.com/gabeur/mmt/blob/master/data_loader/activitynet_dataset.py#L61), they sample a subset from the training data to cross-validate and select model. And in your [implementation](https://github.com/ArrowLuo/CLIP4Clip/blob/508ffa3de39ba0563a03199c440ab602a72e9b6f/dataloaders/data_dataloaders.py#L251), you directly validate on the 1k testing set. Is it ok to use the testing set to select the best model?

**Comment by ArrowLuo** [comment]
Hi @zhixinma, plz find #32.

**Comment by zhixinma** [comment]
Thanks for your prompt reply. I saw the different parameters in [line 494](https://github.com/ArrowLuo/CLIP4Clip/blob/69603645f52788b2a01adffd6428a819849ce2f5/main_task_retrieval.py#L494) and [line 497](https://github.com/ArrowLuo/CLIP4Clip/blob/69603645f52788b2a01adffd6428a819849ce2f5/main_task_retrieval.py#L497). But it seems that the parameter 'subset' is not used in the [test dataloader function](https://github.com/ArrowLuo/CLIP4Clip/blob/69603645f52788b2a01adffd6428a819849ce2f5/dataloaders/data_dataloaders.py#L37) of msrvtt.

**Comment by ArrowLuo** [comment]
Hi @zhixinma, yes. For some datasets, there is no real test dataset. If in such a case, the val dataset is equal to the test dataset in our set. The MSR-VTT is in such a case.

**Comment by zhixinma** [comment]
Hi @ArrowLuo, thanks for your reply! In the aforementioned ECCV'20 paper's [code](https://github.com/gabeur/mmt/blob/master/data_loader/msrvtt_dataset.py#L78), they take the 1k test videos as the real test set. For the missing validation set, they sample a subset from the training data to select the best checkpoint. In your and [Howto100m](https://github.com/antoine77340/howto100m/blob/50af0ffb10ff5d9a1c9c682fa2094151dc852aa7/eval.py#L59)'s implementation, the test set is treated as the validation set. That's why I am confused about the data split.

**Comment by ArrowLuo** [comment]
Hi @zhixinma, thanks for your mention. We did not split extra val in our experiments. You could take our setting if you want to make a comparison with our results. Best~
